### PR TITLE
Update model comparison and SMC notebooks

### PR DESCRIPTION
**Pull request by aloctavodia**
Update notebooks following guidelines in #2834, also fix some typos and update deprecated syntax.

### CODE REVIEW

**Comment by junpenglao** [comment]
LGTM.

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 1]
```diff
@@ -114,20 +73,20 @@
     "    \n",
     "    NOTE: latent_sigma_y is used to create a normally distributed,\n",
     "    'latent error' aka 'inherent noise' in the 'physical process' \n",
-    "    generating thses values, rather than experimental measurement error. \n",
+    "    generating theses values, rather than experimental measurement error. \n",
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb]: these--s--

>    …in the ‘physical' data generating process, rather…

would sound more familiar IMO

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 2]
```diff
@@ -648,41 +571,30 @@
...
-   "source": [
-    "## Create Higher-Order Linear Models"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Back to the real purpose of this Notebook: demonstrate model selection.\n",
+    "Back to the real purpose of this Notebook, demonstrate model selection.\n",
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb]: > , to demonstrate

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 3]
```diff
@@ -1133,8 +1010,7 @@
     "\n",
     "+ Quadratic-generated data (rhs):\n",
     "    + The WAIC is also quite flat across the models\n",
-    "    + The lowest WAIC is model **k4**, but **k3** - **k5** are more or less the same. \n",
-    "    "
+    "    + The lowest WAIC is model **k4**, but **k3** - **k5** are more or less the same. "
    ]
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb:1020]: Looking at the plots, I believe this should be k3 - k**4**

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 4]
```diff
@@ -1224,7 +1093,23 @@
     "\n",
     "It is important to keep in mind that, with more data points, the real underlying model (one that we used to generate the data) should outperforms other models. \n",
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb]: > should outperform--s--

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 5]
```diff
@@ -1224,7 +1093,23 @@
...
+    "In general, PSIS-LOO is recommended. To quote from [avehtari's comment](https://github.com/pymc-devs/pymc3/issues/938#issuecomment-313425552): \"I also recommend using PSIS-LOO instead of WAIC, because it's more reliable and has better diagnostics as discussed in http://link.springer.com/article/10.1007/s11222-016-9696-4 (preprint https://arxiv.org/abs/1507.04544), but if you insist to have one information criterion then leave WAIC\". Alternatively, Watanabe [says](http://watanabe-www.math.dis.titech.ac.jp/users/swatanab/index.html) \"WAIC is a better approximator of the generalization error than the pareto smoothing importance sampling cross validation. The Pareto smoothing cross validation may be the better approximator of the cross validation than WAIC, however, it is not of the generalization error\"."
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb]: > There is some agreement that PSIS-LOO offers the best indication of a model's quality.

File: docs/source/notebooks/GLM-model-selection.ipynb [hunk 6]
```diff
@@ -1071,26 +942,32 @@
...
     "    \n",
-    "In this case we are interested in the WAIC score, we can plot also the standard error of the estimation, which is nice."
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "##### Now loop through all the models and calculate the WAIC"
+    "In this case we are interested in the WAIC score, we can plot also the standard error of the estimation, which is nice.\n",
```
Comment (gBokiau) [inline docs/source/notebooks/GLM-model-selection.ipynb]: > In this case we are interested in the WAIC score. We also plot error bars for the standard error of the estimated scores. This gives us a more accurate view of how much they might differ.

**Review by gBokiau — COMMENTED** [review]
(nitpicking really)

**Comment by aloctavodia** [comment]
thanks @gBokiau for your help. I included almost all your suggestions.

**Pull request merged**